include/Graph/Graph.hpp

```diff
@@ -248,7 +250,9 @@ class Graph {
    * corrispondent to the link
    * Note: No Thread Safe
    */
-  virtual const std::shared_ptr<AdjacencyMatrix<T>> getAdjMatrix() const;
+  virtual std::shared_ptr<AdjacencyMatrix<T>> getAdjMatrix() const;
+
+  virtual std::shared_ptr<AdjacencyMatrix<T>> getCacheAdjMatrix();
```

Comment:
Since this method is used to modify the cached matrix, maybe it would be better to make it `void`

Reply:
Done. Please review.